configs/typescript.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+
+import globals from "globals";
+import javascript from "./javascript";
+import typescriptEslint from "typescript-eslint";
+
+export default [
+  ...javascript,
+  ...typescriptEslint.configs.recommendedTypeChecked,
+  ...typescriptEslint.configs.stylisticTypeChecked,
```

Comment:
I'm not sure if all projects would want these rules from [stylisticTypeChecked](https://typescript-eslint.io/users/configs/#stylistic-type-checked) by default or not. I could imagine that some projects might use Prettier directly, or have some other specific ruleset for their formatting.

What do you think?